### PR TITLE
Minor: add `ExecutionPlan.execute()` snippet for sync methods

### PR DESCRIPTION
**Pull request by comphead**
## Which issue does this PR close?

Closes #.

## Rationale for this change
Very often the question arises on debugging or using record stream in sync context. Adding a code snippet for `ExecutionPlan.execute()`

## What changes are included in this PR?
A code snippet

### CODE REVIEW

File: datafusion/physical-plan/src/lib.rs [hunk 1]
```diff
@@ -235,7 +235,28 @@ pub trait ExecutionPlan: Debug + DisplayAs + Send + Sync {
...
+    ///
+    /// Often its needed to call [`.await`](https://doc.rust-lang.org/std/keyword.await.html) just after the stream async combinators.
+    /// To call it within the sync method please refer the snippet below. The snippet can also be used to help output/debug the node execution on batch level
+    ///
+    /// ```no_run
+    /// use datafusion_common::DataFusionError;
+    ///
+    /// fn print_plan_exec() -> Result<(), DataFusionError> {
+    ///     let stream = node.execute(0, task_ctx)?;
+    ///     futures::stream::once(async move {
```
Comment (tustvold) [inline datafusion/physical-plan/src/lib.rs:249]: This stream gets immediately dropped and therefore will never run?

Comment (comphead) [reply datafusion/physical-plan/src/lib.rs:249]: okay, I'll include another use case with amending stream and returning it back to the user. Like in https://github.com/apache/arrow-datafusion/blob/main/datafusion/physical-plan/src/sorts/sort.rs#L895

@tustvold I believe you mean exactly that?

Comment (tustvold) [reply datafusion/physical-plan/src/lib.rs:249]: I mean as written this example doesn't do anything and is therefore just misleading.

> Like in https://github.com/apache/arrow-datafusion/blob/main/datafusion/physical-plan/src/sorts/sort.rs#L895

Yes, that is I think a good example of the pattern

**Comment by tustvold** [comment]
I tried to encode what I was expecting in https://github.com/apache/arrow-datafusion/pull/8013 PTAL

**Comment by comphead** [comment]
Closed in favor of #8013

**Pull request closed without merging**